ObjCVideoQuickstart/ViewController.m

```diff
@@ -0,0 +1,322 @@
+    }
+    
+    // Disconnect and mic button will be displayed when client is connected to a room.
+    self.disconnectButton.hidden = YES;
+    self.micButton.hidden = YES;
+    
+    self.roomTextField.delegate = self;
+
+    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(dismissKeyboard)];
+    [self.view addGestureRecognizer:tap];
+}
+
+#pragma mark - Public
+
+- (IBAction)connectButtonPressed:(id)sender {
+    [self showRoomUI:YES];
+    [self dismissKeyboard];
+    
+    if ([self.accessToken isEqualToString:@"TWILIO_ACCESS_TOKEN"]) {
+        [self logMessage:[NSString stringWithFormat:@"Fetching an access token"]];
+        [TokenUtils retrieveAccessTokenFromURL:self.tokenUrl completion:^(NSString *token, NSError *err) {
+            dispatch_async(dispatch_get_main_queue(), ^{
+                if (!err) {
+                    self.accessToken = token;
+                    [self doConnect];
+                } else {
+                    [self logMessage:[NSString stringWithFormat:@"Error retrieving the access token"]];
+                    [self showRoomUI:NO];
+                }
+            });
+        }];
+    } else {
+        [self doConnect];
+    }
+}
+
+- (IBAction)disconnectButtonPressed:(id)sender {
+    [self.room disconnect];
+}
+
+- (IBAction)micButtonPressed:(id)sender {
+    // We will toggle the mic to mute/unmute and change the title according to the user action. 
+    
+    if (self.localAudioTrack) {
+        self.localAudioTrack.enabled = !self.localAudioTrack.isEnabled;
+        
+        // Toggle the button title
+        if (self.localAudioTrack.isEnabled) {
+            [self.micButton setTitle:@"Mute" forState:UIControlStateNormal];
+        } else {
+            [self.micButton setTitle:@"Unmute" forState:UIControlStateNormal];
+        }
+    }
+}
+
+#pragma mark - Private
+
+- (void)startPreview {
+    if ([PlatformUtils isSimulator]) {
+        return;
+    }
+    
+    self.camera = [[TVICameraCapturer alloc] init];
+    self.localVideoTrack = [self.localMedia addVideoTrack:YES capturer:self.camera];
+    if (!self.localVideoTrack) {
+        [self logMessage:@"Failed to add video track"];
+    } else {
+        // Attach view to video track for local preview
+        [self.localVideoTrack attach:self.previewView];
+        
+        [self logMessage:@"Video track added to localMedia"];
+        
+        UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self
+                                                                              action:@selector(flipCamera)];
+        [self.previewView addGestureRecognizer:tap];
+    }
+}
+
+- (void)flipCamera {
+    [self.camera flipCamera];
+}
+
+- (void)prepareLocalMedia {
+    
+    // We will offer local audio and video when we connect to room.
+    
+    // Adding local audio track to localMedia
+    self.localAudioTrack = [self.localMedia addAudioTrack:YES];
```

Comment:
@ptankTwilio You should not add the audio track if it has already been added before. Wrap in `if (!self.localAudioTrack) {}`